### ISSUE TITLE
legacy ‘crop and rotate’ has “insane data” on dt 5.0 on Windows

### ISSUE THREAD

**Issue by cabuy** [darktable-org/darktable#18076]
### Describe the bug

“module ‘crop and rotate’ has insane data so it is bypassed for now. you should disable it or change parameters” 
intermittently appears when selecting another picture from the filmstrip in the darkroom

### Steps to reproduce

Doesn’t happen all the time. I have taking a photo of the display with my smartphone to obtain evidence. See attached.
Please advise how I can provide relevant debug data.
Can it be a similar issue to #18029 which seems to be closed ?

### Expected behavior

This message generates some nuisance since it happens on legacy edits (‘crop and rotate’ is a deprecated module) while it never happened before.

### Where did you obtain darktable from?

downloaded from www.darktable.org

### darktable version

5.0

### What OS are you using?

Windows

### What is the version of your OS?

Microsoft Windows 11 Pro

### Describe your system?

Microsoft Windows 11 Pro 23H2 Build 22631.4602
Computer/laptop MSI P65 Creator 9SG
Processor Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz, 2301 MHz, 8 core('s), 16 threads
BIOS American Megatrends Inc. E16Q4IMS.318, 18/05/2020
RAM 32,0 GB
![IMG_20241225_221847325 (002)](https://github.com/user-attachments/assets/af9ccd5e-b80b-4293-b271-4d7f64942dce)

### Are you using OpenCL GPU in darktable?

Yes

### If yes, what is the GPU card and driver?

NVIDIA Geforce RTX 2080 with Max-Q – 8 GB - NVIDIA Studio Driver 566.36

**Comment by pehar1** [comment]
> Can it be a similar issue to https://github.com/darktable-org/darktable/issues/18029 which seems to be closed ?

No. This bug you mention has been fixed (see https://github.com/darktable-org/darktable/pull/18032). Even with this fix for '_retouch_' the message '_insane data_' still appears (comes from '_ashift_').

I also see this messages when reworking old edits from 2020 where the deprecated '_crop and rotate_' has been used. To the best of my knowledge, this message is the result of a security check that was introduced to make the module more stable and avoid crashes.

Since the module has been deprecated for some time, no bug patches are to be expected from the developers. '_crop and rotate_' is a 'code  monster' and its functionality has been splitted into several modules for good reasons.

For affected edits I replace '_crop and rotate_' with '_crop_', '_orientation_' and '_rotate and perspective_' as needed.

**Comment by jenshannoschwalm** [comment]
> To the best of my knowledge, this message is the result of a safety check that was introduced to make the module more stable and avoid crashes.

100% correct. It's about avoiding crashes so if there are insane internal parameters - we had many issues reported - we won't let darktable crash but disable further processing and at least are nice to the user by telling.

@cabuy you might provide raw and xmp, i would have a look if the current check is not good enough ...

**Comment by pehar1** [comment]
[DSC07311.ARW.zip](https://github.com/user-attachments/files/18251153/DSC07311.ARW.zip)
A sample ...

**Comment by jenshannoschwalm** [comment]
Hmm, that one doesn't show the warning here, also processing is all good

**Comment by pehar1** [comment]
For me the message reproducible appears if I double click the image in filmstrip of darkroom, even if the image is already the active one.

**Comment by pehar1** [comment]
Checked again with 5.1.0+9~g71233fe6c4, fresh `.config`, only imported this two samples. Same behaviour, message when double clicking image in filmstrip, no message when clicking through history.

Searching for sample showing message when 'browsing' history ...

**Comment by pehar1** [comment]
Another observation :

1. having two versions of image in duplicate manager
2. history of first image contains _crop and rotate_, edited 2020
3. history of second image (duplicate) has been reworked, _crop_ instead of _crop and rotate_
4. clicking through history of first image containing _crop and rotate_ -> no message
5. double clicking second image (**no _crop and rotate_**) in duplicate manager - > **error message**
6. double clicking first image (**including _crop and rotate_**) in duplicate manager - > **no message**

Steps 5 and 6 can be reproduced repeatedly. Can provide the samples if desired.

**Comment by jenshannoschwalm** [comment]
Yes, it **gets** deactivated if having bad data. Only with filmstrip?

**Comment by pehar1** [comment]
With 'deactivated'  in _duplicate manager_ I mean point 5 and 6 in my previous post:

- first image, **containing** _crop and rotate_ is 'active' (selected), shown for editing in main image area, no warning
- now I select (activate for editing) the second image (duplicate) by double-clicking in _duplicate manager_, second image (**no crop and rotate**) is now shown for editing in main image area and the **warning overlay is shown**.

In filmstrip the simplest way to reproduce is just to double click (select) the already selected image containing _crop and rotate_ again.

**Comment by pehar1** [comment]
Or take one of my samples provided

- goto _duplicate manager_
- press '_duplicate_' 
- see warning

**Comment by pehar1** [comment]
Created a short video.

https://github.com/user-attachments/assets/83a2bb3a-07f7-4d79-85b1-253f4ec82a16

Unfortunately could not get the mouse cursor visible so here a short description :

- both samples in lighttable (fresh `.config`)
- double-click second image -> darkroom -> second image crrectly shown -> no warning
- double-click first image in filmstrip -> first image crrectly shown -> **warning**
- double-click second image in filmstrip -> second image crrectly shown -> **warning**
- goto duplicate manager -> click duplicate -> **warning**

Clicking step by step through the history does not create a warning, processings seems OK. Applies to both images.

**Comment by jenshannoschwalm** [comment]
Yes, i can reproduce here if for some reason very small images are processed like in filmstrip.

I looked up the code again - iirc i did the that - and there is a simple check at the correct position. 
If **for some reason** the area to be processed is too small - we test for width & height being less than 4 - it's not safe to proceed the pixelpipe further so the module get's disabled to avoid a crash later in the pipe.

Why (the for some reason) this is happening at all is deeply hidden in the monster-code and none of the developers was willing /able to maintain that any more, so the module became deprecated and has been replaced by much better and safe options.
It might be a code error or insane parameters, it's as simple as that. You/We have to live with that.

So a clear **won't fix**, sorry.

**Comment by pehar1** [comment]
Absolutely my opion after I had a quick look at ahift.c. Horrible 6000 + lines of code. I already went the way to replace crop and rotate whenever the warning popped up. Thank you for looking at this ...

**Comment by cabuy** [comment]
@jenshannoschwalm and @pehar1 : thank you for your work to reproduce and for your analysis.

**Comment by sylvester328** [comment]
> Absolutely my opion after I had a quick look at ahift.c. Horrible 6000 + lines of code. I already went the way to replace crop and rotate whenever the warning popped up. Thank you for looking at this ...

I think ashift.c is used for 'rotate and perspective'.  The deprecated 'crop and rotate' function is in clipping.c.  I hope I am right as I am looking into changing the LR XMP import to use new modules rather than 'crop and rotate'.

**Comment by pehar1** [comment]
> The deprecated 'crop and rotate' function is in clipping.c

You are absolutely right. My mistake, replace `ashift` with `clipping` and 6000 with 3300 in my previsous comment :-)
